### ISSUE TITLE
No detection on video

### ISSUE THREAD

**Issue by kkt0521** [AlexeyAB/darknet#2406]
I trained weigths with our custom datasets.
We made datasets with yolo_mark.
Training was fine and testing on image file was fine too.
But testing on video file, there were any results.

[Test on Image]
![predictions_retrain2_cap_2](https://user-images.githubusercontent.com/28796035/52095331-ea5a7700-2605-11e9-9aa8-ba2fa5dd326b.jpg)

[Test on Video]
![test_on_video](https://user-images.githubusercontent.com/28796035/52095575-d6fbdb80-2606-11e9-95e4-cef4891c69e4.JPG)

I use yolo-obj.cfg file made by Yolo-Mark.

<yolo-obj.cfg>
[net]

# Testing
batch=1
subdivisions=1

# Training
#batch=256
#subdivisions=64
height=416
width=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.0001
max_batches = 45000
policy=steps
steps=100,25000,35000
scales=10,.1,.1

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=64
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=128
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

#######

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[route]
layers=-9

[reorg]
stride=2

[route]
layers=-1,-3

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=45
activation=linear

[region]
anchors = 1.08,1.19, 3.42,4.41, 6.63,11.38, 9.42,5.11, 16.62,10.52
bias_match=1
classes=4
coords=4
num=5
softmax=1
jitter=.2
rescore=1

object_scale=5
noobject_scale=1
class_scale=1
coord_scale=1

## absolute=1
thresh = .6
random=0
I use the command like these.

(1)darknet.exe detector test data/obj_re.data yolo-obj.cfg yolo-obj_retrain_0123.weights data/obj/vv248.jpg

(2)darknet.exe detector demo data/obj_re.data yolo-obj.cfg yolo-obj_retrain_0123.weights test.mp4

I use two commands for testing image and video.

Is there any problem?

I think it can't find objects with trained weights.
Then, Why testing on image was done???

Please, notice me.

**Comment by kkt0521** [comment]
And it works in video with your yolo-voc.weights in same code.

**Comment by AlexeyAB** [comment]
> No detection on video

May be objects on video are very different from object in Training dataset. Try to follow these rules during training: https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

Also try to use this command:

`darknet.exe detector demo data/obj_re.data yolo-obj.cfg yolo-obj_retrain_0123.weights test.mp4 -thresh 0.05`

**Comment by kkt0521** [comment]
Above the video is one of  the training datasets. I divided video into several frames and used each frames.
And your suggest command is not working.
There's no output.
I will remake the cfg file with your rules and upload the output lines.
